alternator/executor.cc

```diff
@@ -2795,16 +2798,22 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
             if (r_name == "PutRequest") {
                 const rjson::value& put_request = r->value;
                 const rjson::value& item = put_request["Item"];
-                mutation_builders.emplace_back(schema, put_or_delete_item(
+                auto&& put_item = put_or_delete_item(
                         item, schema, put_or_delete_item::put_item{},
-                        si_key_attributes(_proxy.data_dictionary().find_table(schema->ks_name(), schema->cf_name()))));
+                        si_key_attributes(_proxy.data_dictionary().find_table(schema->ks_name(), schema->cf_name())));
+                auto half_units = wcu_consumed_capacity_counter::get_half_units(put_item.length_in_bytes());
+                _stats.wcu_total[stats::PUT_ITEM] += half_units;
```

Comment:
I don't know where my comment here went (I guess it disappeared when you changed the name of the variable so github doesn't know it's the same code), so I'll write it again here:

At this point, if I understand correctly, you're only **parsing** the request. It is still possible that as we continue to parse the request we'll find some silly error (e.g., non-existent table name, or incorrect parameters, or something) and we will not perform any write at all. So it is premature to increment _stats.wcu_total here.

It's not a big bug - it will not affect correct requests at all - but I still think it's a bug.